### PR TITLE
[FLINK-33853][runtime][JUnit5 Migration] Migrate Junit5 for DeclarativeSlotPoolBridge test classes of runtime module

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

[FLINK-33853][runtime][JUnit5 Migration] Migrate Junit5 for DeclarativeSlotPoolBridge test classes of runtime module

## Brief change log

- DeclarativeSlotPoolBridgeRequestCompletionTest
- DeclarativeSlotPoolBridgeResourceDeclarationTest
- DeclarativeSlotPoolBridgePreferredAllocationsTest
- DeclarativeSlotPoolBridgeTest

## Verifying this change

N.A

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by RocMarshal** [comment]
Hi, @1996fanrui Could you help review this pr if you're in the free time?
Thank you~

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgePreferredAllocationsTest.java [hunk 1]
```diff
@@ -45,10 +45,10 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 @ExtendWith(TestLoggerExtension.class)
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgePreferredAllocationsTest.java:47]:
```suggestion
```

`TestLoggerExtension` has been added in the module level.

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeRequestCompletionTest.java [hunk 2]
```diff
@@ -43,27 +44,24 @@
...
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.containsInAnyOrder;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.not;
-import static org.hamcrest.Matchers.nullValue;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Tests how the {@link DeclarativeSlotPoolBridge} completes slot requests. */
-public class DeclarativeSlotPoolBridgeRequestCompletionTest extends TestLogger {
+@ExtendWith(TestLoggerExtension.class)
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeRequestCompletionTest.java:50]:
```suggestion
```

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeResourceDeclarationTest.java [hunk 3]
```diff
@@ -44,37 +45,33 @@
...
 import static org.apache.flink.runtime.jobmaster.slotpool.DeclarativeSlotPoolBridgeTest.createAllocatedSlot;
 import static org.apache.flink.runtime.jobmaster.slotpool.DeclarativeSlotPoolBridgeTest.createDeclarativeSlotPoolBridge;
-import static org.hamcrest.CoreMatchers.is;
-import static org.junit.Assert.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Tests for the {@link DeclarativeSlotPoolBridge}. */
-@RunWith(Parameterized.class)
-public class DeclarativeSlotPoolBridgeResourceDeclarationTest extends TestLogger {
+@ExtendWith({TestLoggerExtension.class, ParameterizedTestExtension.class})
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeResourceDeclarationTest.java:54]:
```suggestion
@ExtendWith(ParameterizedTestExtension.class)
```

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeResourceDeclarationTest.java [hunk 4]
```diff
@@ -141,40 +137,38 @@ public void testRequirementsDecreasedOnAllocationTimeout() throws Exception {
                             .get();
 
             // waiting for the timeout
-            assertThat(
-                    allocationFuture,
-                    FlinkMatchers.futureWillCompleteExceptionally(Duration.ofMinutes(1)));
-
+            assertThatFuture(allocationFuture).failsWithin(Duration.ofMinutes(1));
+            Thread.sleep(5L);
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeResourceDeclarationTest.java:141]: Why is it needed here?

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeTest.java [hunk 5]
```diff
@@ -55,34 +55,30 @@
...
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.Assert.fail;
+import static org.apache.flink.core.testutils.FlinkAssertions.assertThatFuture;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /** Tests for the {@link DeclarativeSlotPoolBridge}. */
-@RunWith(Parameterized.class)
-public class DeclarativeSlotPoolBridgeTest extends TestLogger {
+@ExtendWith({TestLoggerExtension.class, ParameterizedTestExtension.class})
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeTest.java:63]:
```suggestion
@ExtendWith(ParameterizedTestExtension.class)
```

**Review by 1996fanrui — COMMENTED** [review]
Thanks @RocMarshal for the contribution!

I left some minor comments, please take a look in your free time, thanks~

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeResourceDeclarationTest.java (hunk 4 above)
Comment (RocMarshal) [reply flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeResourceDeclarationTest.java:141]: Nice catch! 
sorry for it. Maybe it's a dirty line from the cp.
I'll remove it and check it on the other corresponding PRs.

**Review by 1996fanrui — APPROVED** [review]
Thanks @RocMarshal for the update!

LGTM assuming CI is green.

**Pull request merged**